errors/errors.go

```diff
@@ -47,19 +47,23 @@ func (p *Error) StackString() string {
 
 // VerboseString returns the error message, stack trace and contexts
 func (p *Error) VerboseString() string {
-	return fmt.Sprintf("%s\nPublic context: %+v\nPrivate context: %+v\n%s", p.Error(), p.PublicContext, p.PrivateContext, p.StackString())
+	return fmt.Sprintf("%s\nParams: %+v\n%s", p.Error(), p.Params, p.StackString())
 }
 
 // New creates a new error for you. Use this if you want to pass along a custom error code.
 // Otherwise use the handy shorthand factories below
-func New(code int, message string, context ...map[string]string) *Error {
-	return errorFactory(code, message, context...)
+func New(code string, message string, params map[string]string) *Error {
+	return errorFactory(code, message, params)
 }
 
 // Wrap takes any error interface and wraps it into an Error.
 // This is useful because an Error contains lots of useful goodies, like the stacktrace of the error.
 // NOTE: If `err` is already an `Error` the passed contexts will be ignored
-func Wrap(err error, context ...map[string]string) *Error {
+func Wrap(err error, params map[string]string) *Error {
```

Comment:
http://golang.org/doc/faq#nil_error

"It's a good idea for functions that return errors always to use the error type in their signature (as we did above) rather than a concrete type such as *MyError, to help guarantee the error is created correctly. As an example, os.Open returns an error even though, if not nil, it's always of concrete type *os.PathError."